hedera-node/docs/design/services/addressbook-service/addressbook-service.md

```diff
@@ -0,0 +1,143 @@
+# Addressbook Service
+
+The Address Book Service is responsible for configuring the list of nodes on the network.
+Nodes can be added, updated or removed via the Address Book Service and changes take effect once the
+network is restarted. This allows the Hedera council to add and remove nodes, and node operators to configure nodes
+on the network.
+
+### Table of Contents
+- [Addressbook Service](#Addressbook-Service)
+- [Table of Contents](#Table-of-Contents)
+- [Architecture Overview](#Architecture-Overview)
+- [Protobuf Definitions](#Protobuf-Definitions)
+  - [Transactions for the Addressbook Service](#Transactions-for-the-Addressbook-Service)
+  - [NodeCreateTransactionBody](#NodeCreateTransactionBody)
+  - [NodeUpdateTransactionBody](#NodeUpdateTransactionBody)
+  - [NodeDeleteTransactionBody](#NodeDeleteTransactionBody)
+- [Addressbook Service Handlers](#Addressbook-Service-Handlers)
+- [Network Response Messages](#Network-Response-Messages)
+
+## Key Features of the Addressbook Service
+The Addressbook Service is a service (```AddressBookService extends RpcService```) that provides a decentralized and secure
+way to update the node addressbook to configure nodes on the Hedera network. It allows the Hedera council to add and
+remove nodes, and node operators to configure nodes on the network. Changes are saved to state. When the network receives
+a freeze transactions, changes are persisted to disk in a file `config.txt`. 
+Changes take effect once the network is restarted.
+
+## Architecture Overview
+
+The Addressbook Service is designed to handle transactions and queries related to the dynamic addressbook that specifies
+nodes on the Hedera network. It provides a set of operations that allow users to add, delete and update node information, 
+including consensus staking. The main components of the Addressbook Service are:
```

Comment:
including consensus staking?

Reply:
That's part of the next phase of the project. I will delete it here, since the rest of the documentation reflects what we've implemented for phase 2.